frank-flow/src/frontend/src/app/shared/components/edit-dialog/edit-dialog.component.html

```diff
@@ -0,0 +1,27 @@
+<ngx-smart-modal
+  #editDialog
+  (onDataAdded)="onDataAdded()"
+  (onOpenFinished)="inputElement.focus()"
+  (onAnyCloseEvent)="edit()"
+  class="editDialog"
+  customClass="nsm-centered"
+  identifier="editDialog"
+>
+  <div>
+    <h2>
+      Modify {{ currentDirectory?.path ? 'folder' : 'file' }} {{ oldFileName }}
+    </h2>
+  </div>
+  <label for="inputElement">Name</label>
+  <input
+    #inputElement
+    [(ngModel)]="fileName"
+    type="text"
+    (keyup.enter)="editDialog.close()"
+  />
+  <label for="deleteButton"
+    >By clicking this button, the
+    {{ currentDirectory?.path ? 'folder' : 'file' }} wil be permanantly deleted
+  </label>
+  <app-button #deleteButton>Delete</app-button>
```

Comment:
Place in one row (with input-row).
Remove the element reference.
Hookup the remove method to the button with `(click)`